src/osal/lv_sdl2.c

```diff
@@ -0,0 +1,171 @@
+/**
+ * @file lv_sdl2.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "lv_os.h"
+
+#if LV_USE_OS == LV_OS_SDL2
+
+#include <errno.h>
+#include "../misc/lv_log.h"
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+static int generic_callback(void * user_data);
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+lv_result_t lv_thread_init(lv_thread_t * thread, lv_thread_prio_t prio, void (*callback)(void *), size_t stack_size,
+                           void * user_data)
+{
+    LV_UNUSED(prio);
+    thread->callback = callback;
+    thread->user_data = user_data;
+    thread->thread = SDL_CreateThreadWithStackSize(generic_callback, "LVGL", stack_size, thread);
+    return LV_RESULT_OK;
+}
+
+lv_result_t lv_thread_delete(lv_thread_t * thread)
+{
+    int ret;
+    SDL_WaitThread(thread->thread, &ret);
+    if(ret != 0) {
+        LV_LOG_WARN("Error: %d", ret);
+        return LV_RESULT_INVALID;
+    }
+
+    return LV_RESULT_OK;
+}
+
+lv_result_t lv_mutex_init(lv_mutex_t * mutex)
+{
+    *mutex = SDL_CreateMutex();
+
+    if(*mutex == NULL) {
+        LV_LOG_WARN("Error: %s", SDL_GetError());
+        return LV_RESULT_INVALID;
+    }
+    else {
+        return LV_RESULT_OK;
+    }
+}
+
+lv_result_t lv_mutex_lock(lv_mutex_t * mutex)
+{
+    int ret = SDL_LockMutex(*mutex);
+    if(ret) {
+        LV_LOG_WARN("Error: %d", ret);
+        return LV_RESULT_INVALID;
+    }
+    else {
+        return LV_RESULT_OK;
+    }
+}
+
+lv_result_t lv_mutex_lock_isr(lv_mutex_t * mutex)
+{
+    int ret = SDL_LockMutex(*mutex);
+    if(ret) {
+        LV_LOG_WARN("Error: %d", ret);
+        return LV_RESULT_INVALID;
+    }
+    else {
+        return LV_RESULT_OK;
+    }
+}
+
+lv_result_t lv_mutex_unlock(lv_mutex_t * mutex)
+{
+    int ret = SDL_UnlockMutex(*mutex);
+    if(ret) {
+        LV_LOG_WARN("Error: %d", ret);
+        return LV_RESULT_INVALID;
+    }
+    else {
+        return LV_RESULT_OK;
+    }
+}
+
+lv_result_t lv_mutex_delete(lv_mutex_t * mutex)
+{
+    SDL_DestroyMutex(*mutex);
+    return LV_RESULT_OK;
+}
+
+lv_result_t lv_thread_sync_init(lv_thread_sync_t * sync)
+{
+    sync->mutex = SDL_CreateMutex();
+    sync->cond = SDL_CreateCond();
```

Comment:
Hmm different from other objects you are not testing they are valid after allocating the, they will be always valid, otherwise, can you validate their creation before moving forward?

Reply:
No problem, I created it using lv_pthread.c as a template and didn't add any extra logic，i also think that check result is necessary, but I'm not sure if there is another reason why lv_pthread.c & lv_windows.c doesn't do this.